### PR TITLE
[Frontend] MQLLMEngine supports profiling.

### PR DESCRIPTION
**Pull request by Abatom**
MQLLMEngine supports profiling.

1. The profiling-related protocols and APIs from previous versions have been adapted to MQLLMEngine.

### CODE REVIEW

File: vllm/engine/multiprocessing/client.py [hunk 1]
```diff
@@ -497,3 +497,17 @@ async def _process_request(
                     await self.abort(request_id)
         finally:
             self.output_queues.pop(request_id)
+
+    async def start_profile(self) -> None:
+        """Start profiling the engine"""
+
+        await self._send_one_way_rpc_request(
+            request=RPCUProfileRequest.START_PROFILE,
+            socket = self.input_socket)
```
Comment (njhill) [inline vllm/engine/multiprocessing/client.py:506]: nit: formatting
```suggestion
            socket=self.input_socket)
```

File: vllm/engine/multiprocessing/client.py [hunk 2]
```diff
@@ -497,3 +497,17 @@ async def _process_request(
...
+
+    async def stop_profile(self) -> None:
+        """Stop profiling the engine"""
+
+        await self._send_one_way_rpc_request(
+            request=RPCUProfileRequest.STOP_PROFILE,
+            socket = self.input_socket)
```
Comment (njhill) [inline vllm/engine/multiprocessing/client.py:513]:
```suggestion
            socket=self.input_socket)
```

**Review by njhill — COMMENTED** [review]
Thanks @abatom, looks great! Could you run `./format.sh` to sure the formatting is correct.

**Comment by abatom** [comment]
# ./format.sh 
vLLM yapf: Done
vLLM mypy:
Success: no issues found in 122 source files
Success: no issues found in 294 source files
Success: no issues found in 15 source files
Success: no issues found in 12 source files
Success: no issues found in 17 source files
Success: no issues found in 17 source files
Success: no issues found in 17 source files
Success: no issues found in 87 source files
Success: no issues found in 6 source files
Success: no issues found in 15 source files
Success: no issues found in 22 source files
vLLM mypy: Done
vLLM codespell: Done
All checks passed!
vLLM ruff: Done
vLLM isort: Done

File: vllm/engine/multiprocessing/engine.py [hunk 3]
```diff
@@ -356,6 +363,18 @@ def _set_errored(self, e: BaseException):
     def _alive(self):
         self._last_alive_time = time.time()
 
+    def start_profile(self) -> None:
+        if type(self.engine.model_executor) is GPUExecutorAsync:
```
Comment (njhill) [inline vllm/engine/multiprocessing/engine.py:367]:
```suggestion
        if type(self.engine.model_executor) is GPUExecutor:
```

File: vllm/engine/multiprocessing/engine.py [hunk 4]
```diff
@@ -356,6 +363,18 @@ def _set_errored(self, e: BaseException):
...
+            self.engine.model_executor.start_profile()
+        else:
+            self.engine.model_executor._run_workers("start_profile")
+
+    def stop_profile(self) -> None:
+        if type(self.engine.model_executor) is GPUExecutorAsync:
```
Comment (njhill) [inline vllm/engine/multiprocessing/engine.py:373]:
```suggestion
        if type(self.engine.model_executor) is GPUExecutor:
```

**Review by njhill — CHANGES_REQUESTED** [review]
@abatom it won’t be the async version of the executor here

**Comment by abatom** [comment]
@njhill What do I need to do next?

**Review by njhill — APPROVED** [review]
Thanks @abatom

**Comment by abatom** [comment]
@njhill, what should I do for "Required statuses must pass before merging" ?

**Comment by njhill** [comment]
@abatom these are unrelated CI test failures, we can override to merge this.

**Pull request merged**